### PR TITLE
Add range support for Google Sheets

### LINKED ISSUE #3989: bigquery: confirm federated sheets support range property
For some time, the BigQuery backend has supported an additional "range" property when defining a federated table against Google Drive (via GoogleSheetsOptions).  Please verify this property is available to users of the library.  With the next discovery release, the "beta" label for this functionality will be removed.

Docstring for the range field:
[Optional] Range of a sheet to query from. Only used when non-empty.
Typical format: sheet_name!top_left_cell_id:bottom_right_cell_id
For example: sheet1!A1:B20

Contact: shollyman

### PR DESCRIPTION
**Pull request by blowmage**
[closes #3989]

### CODE REVIEW

File: google-cloud-bigquery/lib/google/cloud/bigquery/external.rb [hunk 1]
```diff
@@ -1224,6 +1224,52 @@ def skip_leading_rows= row_count
             frozen_check!
             @gapi.google_sheets_options.skip_leading_rows = row_count
           end
+
+          ##
+          # Range of a sheet to query from. Only used when non-empty. Typical
+          # format: `{sheet_name}!{top_left_cell_id}:{bottom_right_cell_id}`.
+          #
+          # @return [Integer]
```
Comment (quartzmo) [inline google-cloud-bigquery/lib/google/cloud/bigquery/external.rb]: What is the integer that is returned? Can you add a description here?

Comment (quartzmo) [reply google-cloud-bigquery/lib/google/cloud/bigquery/external.rb]: Maybe this is a c/p error and it should be `String`?

File: google-cloud-bigquery/lib/google/cloud/bigquery/external.rb [hunk 2]
```diff
@@ -1224,6 +1224,52 @@ def skip_leading_rows= row_count
...
+          def range
+            @gapi.google_sheets_options.range
+          end
+
+          ##
+          # Set the range of a sheet to query from. Only used when non-empty.
+          # Typical format:
+          # `{sheet_name}!{top_left_cell_id}:{bottom_right_cell_id}`.
+          #
+          # @param [Integer] new_range New range value
```
Comment (quartzmo) [inline google-cloud-bigquery/lib/google/cloud/bigquery/external.rb]: `String`?

File: google-cloud-bigquery/lib/google/cloud/bigquery/external.rb (hunk 1 above)
Comment (blowmage) [reply google-cloud-bigquery/lib/google/cloud/bigquery/external.rb]: Fixed.

File: google-cloud-bigquery/lib/google/cloud/bigquery/external.rb (hunk 2 above)
Comment (blowmage) [reply google-cloud-bigquery/lib/google/cloud/bigquery/external.rb]: Fixed.

**Comment by blowmage** [comment]
@shollyman I should have added you as a reviewer. Are there any changes to this that you want made before releasing?

**Pull request merged**